### DISCUSSION TITLE
Warn when a config value matches the rule's default option value

### DISCUSSION THREAD

**Discussion by scottohara** [eslint/eslint#19182] [Ideas]
I was very excited to see #17448 implemented in v9.15.0.

I currently manage a sizeable eslint config that is intentionally _verbose_  (by which I mean: the config explicitly lists out all options of a given rule, even when a value in the config matches the default value for the rule).

For example, my `accessor-pairs` config is currently:
```js
"accessor-pairs": [
  "error",
  {
    setWithoutGet: true,           // unnecessary as the rule default is already true
    getWithoutSet: false,          // unnecessary as the rule default is already false
    enforceForClassMembers: true,  // unnecessary as the rule default is already true
  },
],
```

Which could equally be expressed as simply:

```js
"accessor-pairs": "error"
```

For a long time my intention has been to review & trim down the config, by methodically going through every single rule and removing any such redundant default values.  The resulting config should only contain options that intentionally override/diverge from the rule defaults.
This config would be significantly leaner, easier to maintain, and much easier to see at a glance which rules have custom configuration.

With #17448, I expect this task to be greatly simplified, as it should now be possible to write a small script that programatically compares the config of a given rule against its `defaultOptions` and highlight any that are candidates for review and possible removal.

One complication is that there are sometimes cases where one config object overrides the rule default, and a subsequent later config object restores the default.  In this case, we _wouldn't_ want to suggest removing the latter.  How feasible that would be, I'm not sure.  Example:

```js
{
  rules: {
    "accessor-pairs": ["error", { setWithoutGet: false }]  // overrides rule default
  }
},
{
  files: ["special/path/**/*"],
  rules: {
    "accessor-pairs": ["error", { setWithoutGet: true }]  // restores rule default
  }
}
``` 

Given that rule defaults can & may change over time, this would potentially be something that you would want to repeat periodically.
Which got me thinking, if it is indeed possible to reliably detect when config value can be safely dropped in lieu of the default option, could eslint itself (optionally) emit these warnings?

Alternatively, could this be something added to the config-inspector, e.g. a new indicator and/or filter for rule options in the fully resolved config that could potentially be candidates for removal?

**Comment by scottohara** [comment]
I have further elaborated on the idea of this being an enhancement to the config-inspector:
https://github.com/eslint/config-inspector/discussions/108

**Reply by nzakas** [reply]
Yeah, I think this is the right way to go for this feature.

**Comment by scottohara** [comment]
Merged in https://github.com/eslint/config-inspector/pull/110

🎉
